projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-conditional-filter.component.ts

```diff
@@ -41,7 +41,7 @@ export class IgxExcelStyleConditionalFilterComponent implements OnDestroy {
     public subMenu: IgxDropDownComponent;
 
     protected get filterNumber() {
-        return this.esf.expressionsList.length;
+        return this.esf.expressionsList.filter(e => e.expression.fieldName === this.esf.column.field && e.expression.condition).length;
```

Comment:
Shouldn't the expressionsList always be filterted with these conditions (meaning that we filter on initializing it) instead of having to add the same conditions in a few places?